### PR TITLE
Adding Multipath test

### PR DESCRIPTION
**Pull request by narasimhan-v**
Tests Flushing, Blacklisting, patch_checker policy.

Signed-off-by: Narasimhan V <sim@linux.vnet.ibm.com>

### CODE REVIEW

**Comment by narasimhan-v** [comment]
[job_log.txt](https://github.com/avocado-framework-tests/avocado-misc-tests/files/538772/job_log.txt)

**Comment by narasimhan-v** [comment]
Depends on https://github.com/avocado-framework/avocado/pull/1547

File: io/disk/multipath_test.py [hunk 1]
```diff
@@ -0,0 +1,128 @@
...
+
+class MultipathTest(Test):
+    """
+    Multipath Test functions.
+    """
+    def setUp(self):
+        """
+        Set up.
+        """
+        # Check if multipath devices are present in sytem
```
Comment (PraveenPenguin) [inline io/disk/multipath_test.py]: typo miss `system` please correct

File: io/disk/multipath_test.py [hunk 2]
```diff
@@ -0,0 +1,128 @@
...
+            pkg_name += "device-mapper-multipath"
+        smm = SoftwareManager()
+        if not smm.check_installed(pkg_name) and not smm.install(pkg_name):
+            self.skip("Can not install %s" % pkg_name)
+
+        # Take a backup of current config file
+        self.mpath_file = "/etc/multipath.conf"
+        if os.path.isfile(self.mpath_file):
+            cmd = "cp %s %s.bkp" % (self.mpath_file, self.mpath_file)
+            process.system(cmd)
```
Comment (PraveenPenguin) [inline io/disk/multipath_test.py]: cp command not looks good can you please use `shutil` utility 

code can be 

`shutil.copyfile('/etc/multipath.conf','/etc/multipath.conf.bkp')`

File: io/disk/multipath_test.py [hunk 3]
```diff
@@ -0,0 +1,128 @@
...
+        # Find all details of multipath devices
+        for wwid in self.wwids:
+            self.mpath_dic = {}
+            self.mpath_dic["wwid"] = wwid
+            self.mpath_dic["name"] = multipath.get_mpath_name(wwid)
+            self.mpath_dic["paths"] = multipath.get_paths(wwid)
+            self.mpath_dic["policy"] = multipath.get_policy(wwid)
+            self.mpath_dic["size"] = multipath.get_size(wwid)
+            self.mpath_list.append(self.mpath_dic)
+        pprint(self.mpath_list)
```
Comment (PraveenPenguin) [inline io/disk/multipath_test.py]: ok

File: io/disk/multipath_test.py [hunk 4]
```diff
@@ -0,0 +1,128 @@
...
+                cmd = "devnode %s" % disk
+                multipath.form_conf_mpath_file(blacklist=cmd)
+                if disk in multipath.get_paths(path_dic["wwid"]):
+                    msg += "Blacklist of %s fails\n" % disk
+
+        # Print errors
+        if msg:
+            msg = "Following Tests Failed\n" + msg
+            self.log.debug(msg)
+            self.fail("Some tests failed")
```
Comment (PraveenPenguin) [inline io/disk/multipath_test.py]: This message is misleading.
can you please more elaborate ?

File: io/disk/multipath_test.py [hunk 5]
```diff
@@ -0,0 +1,128 @@
...
+
+    def tearDown(self):
+        """
+        Restore config file, if existed, and restart services
+        """
+        if os.path.isfile(self.mpath_file):
+            cmd = "cp %s.bkp %s" % (self.mpath_file, self.mpath_file)
+            process.run(cmd)
```
Comment (PraveenPenguin) [inline io/disk/multipath_test.py]: here also please use `shutil` utility

File: io/disk/multipath_test.py (hunk 2 above)
Comment (narasimhan-v) [inline io/disk/multipath_test.py]: Will change.

File: io/disk/multipath_test.py (hunk 4 above)
Comment (narasimhan-v) [inline io/disk/multipath_test.py]: `msg` has all the needed info.

File: io/disk/multipath_test.py (hunk 5 above)
Comment (narasimhan-v) [inline io/disk/multipath_test.py]: Sure

File: io/disk/multipath_test.py (hunk 1 above)
Comment (narasimhan-v) [inline io/disk/multipath_test.py]: Sure

**Comment by narasimhan-v** [comment]
Created a v2 of this PR #128

**Pull request closed without merging**